src/utils/getExportTypes.ts

```diff
@@ -235,7 +236,13 @@ const getExportTypes = ({ results, exportId }: Props): ExportTypes => {
             uid,
           }))
       )
-        .filter(({ text }) => matchDiscourseNode({ title: text, ...n }))
+        .filter(({ text }) => {
+          if (matchedTexts.has(text)) return false;
+          const isMatch = matchDiscourseNode({ title: text, ...n });
+          if (isMatch) matchedTexts.add(text);
+
+          return isMatch;
+        })
```

Comment:
What is the purpos of the original matchDiscourseNode filter?
I'm likely missing something, because for the `Export Discourse Graph` command, it seems to just be double up the work?

Reply:
Ah, to get the type maybe?

It seems like an expensive loop (`allNodes` mapped over all results running `matchDiscourseNode` which hits `roamAlphaAPI`) when we I believe we have the type at initialization of `Export Discourse Graph`.

Reply:
I made some changes to pass the `type` with Export Discourse Graph `results`: https://github.com/RoamJS/query-builder/pull/205/commits/34f2311fcc9d005587762e0a78ade17b8d536efc so we can bypass this loop altogether.

Reply:
https://www.loom.com/share/ddd3946b3db94545b89ffa73bfa14aee

Reply:
yea I think is to get the discourse type - would definitely prefer bypassing if we could